### ISSUE TITLE
OpenDataHub operator redeployment fails when Operator Uninstallation is executed

### ISSUE THREAD

**Issue by dmarcus-wire** [opendatahub-io/opendatahub-operator#203]
**Describe the bug**
Logically, clicking "Uninstall Operator" for Open Data Hub should cleanup all the artifacts left behind. It does not and causes errors on installation on the same cluster if you try again. 

**To Reproduce**
1. Go to 'Install Operators' in your custom project i.e. 'odh'
2. Click on 'Open Data Hub Operator'
3. Scroll down to 'Open Data Hub API'
4. Select 'delete kfdef'
5. Uninstall the Open Data Hub Operator entirely
6. Delete the project i.e. 'odh'
7. run the 'oc get' from this gist....https://gist.github.com/dmarcus-wire/21101c8bff798220ad6f676fbb6a4570

```
oc get operatorgroup -n openshift-operators opendatahub

oc get kfdef opendatahub -A

oc get all --selector opendatahub.io/component=true -A

oc get all --selector application-crd-id=data-science-pipelines -A

oc get all --selector app=odh-dashboard -A

oc get all --selector app=odh-model-controller -A

oc get all --selector name=modelmesh-serving-triton-2.x -A

oc get all --selector control-plane=modelmesh-controller -A

oc get all --selector name=modelmesh-serving-ovms-1.x -A

oc get all --selector modelmesh-service=modelmesh-serving -A

oc get all --selector app=model-mesh --all-namespaces

oc get all,configmap,pvc,serviceaccount,rolebinding --selector=application-crd-id=data-science-pipelines --all-namespaces

oc get all,configmap,pvc,serviceaccount,rolebinding --selector=opendatahub.io/component=true --all-namespaces
```

**Expected behavior**
Either the operator or documentation provides instructions for cleaning up ODH entirely and successfully re-deploying

**Comment by dmarcus-wire** [comment]
This script helps with cleanup of ODH 1.4.1 on OCP 4.11.8. https://github.com/redhat-na-ssa/demo-odh

Can this be reviewed and considered for cleanup?

**Comment by VaishnaviHire** [comment]
@dmarcus-wire This issue is resolved partially in 1.5 release. We add KfDef as [ownerreference](https://github.com/opendatahub-io/opendatahub-operator/pull/211) to the application deployments.  This allows cleaning up of application resources when KfDef is deleted. The only exception is deployments created by other operators (e.g prometheus).

ODH operator clean-up process is the same as documented in [operator-sdk docs](https://olm.operatorframework.io/docs/tasks/uninstall-operator/). The following steps should allow reinstall
- Delete KfDef CRs
- Delete ODH namespace
- Uninstall operator
- Reinstall ODH operator and deployments

Is the recommendation to automate the above steps when the user wants to clean up after uninstall? To make sure we test this for 1.5 release can you share the following:
- What errors did you get when reinstalling? 
- Were those related to ODH applications or the deployment of the operator itself?

**Comment by VaishnaviHire** [comment]
No longer an issue with v2 operator
